src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -92,10 +108,49 @@ public Future<Void> updatePiece(Piece pieceToUpdate, boolean createItem, boolean
   }
 
   protected Future<Void> updatePoLine(PieceUpdateHolder holder, RequestContext requestContext) {
-    if (!Boolean.TRUE.equals(holder.getOriginPoLine().getIsPackage()) && !Boolean.TRUE.equals(holder.getOriginPoLine().getCheckinItems())) {
-      return updatePoLineService.updatePoLine(holder, requestContext);
-    }
-    return Future.succeededFuture();
+    CompositePoLine originPoLine = holder.getOriginPoLine();
+
+    return getPiecesByPoLine(originPoLine.getId(), requestContext)
+      .compose(pieces -> {
+        CompositePurchaseOrder order = holder.getOriginPurchaseOrder();
+        if (order.getOrderType() != OrderType.ONE_TIME || order.getWorkflowStatus() != WorkflowStatus.OPEN) {
+          return Future.succeededFuture();
+        }
+        logger.info(">>> " + pieces);
+        List<Piece> piecesToUpdate = List.of(holder.getPieceToUpdate());
+        logger.info(">>> " + calculatePoLineReceiptStatus(originPoLine, pieces, piecesToUpdate));
+        return Future.succeededFuture();
+      }).compose(t -> {
+        if (!Boolean.TRUE.equals(originPoLine.getIsPackage()) &&
+          !Boolean.TRUE.equals(originPoLine.getCheckinItems())) {
+          return updatePoLineService.updatePoLine(holder, requestContext);
+        }
+        return Future.succeededFuture();
+      });
+  }
+
+  PoLine.ReceiptStatus calculatePoLineReceiptStatus(CompositePoLine poLine, List<Piece> fromStorage, List<Piece> toUpdate) {
+
+    // 1. collect all piece statuses
+    Map<String, Piece.ReceivingStatus> map = new HashMap<>();
+    fromStorage.forEach(piece -> map.put(piece.getId(), piece.getReceivingStatus()));
+    toUpdate.forEach(piece -> map.put(piece.getId(), piece.getReceivingStatus()));
+
+    // 2. count received and expected statuses
+    long receivedQuantity = map.values().stream().filter(RECEIVED_STATUSES::contains).count();
+    long expectedQuantity = map.values().stream().filter(EXPECTED_STATUSES::contains).count();
+
+    logger.info("calculatePoLineReceiptStatus:: POL: {}, received: {}, expected: {}",
+      poLine.getId(), receivedQuantity, expectedQuantity);
+
+    return expectedQuantity == 0 ? FULLY_RECEIVED :
+      receivedQuantity > 0 ? PARTIALLY_RECEIVED : AWAITING_RECEIPT;
```

Comment:
sonar says that this statement can be simplified, if possible better to do it, maybe use if statemens instead

Reply:
sonar says that trinary operator is hard to understand and should be replaced with if statements. I can change if you think it is.

Reply:
because for me it's quite natural that we return `FULLY_RECEIVED` when nothing left and `PARTIALLY_RECEIVED` or `AWAITING_RECEIPT` depending on anything received or not.